app/hotels/src/filter/price/PriceFilter.js

```diff
@@ -1,169 +1,105 @@
-      isPopupOpen: true,
-    });
+  function openPopup() {
+    setIsPopupOpen(true);
+  }
 
-  closePopup = (callback?: Function) =>
-    this.setState(
-      {
-        isPopupOpen: false,
-      },
-      callback,
-    );
+  function closePopup() {
+    setIsPopupOpen(false);
+  }
 
-  handleSave = ({
+  function handleSave({
     minPrice,
     maxPrice,
   }: {
     minPrice: number,
     maxPrice: number,
-  }) => {
+  }) {
     const filter = {
-      minPrice:
-        minPrice !== this.props.currentSearchStats.priceMin ? minPrice : null,
-      maxPrice:
-        maxPrice !== this.props.currentSearchStats.priceMax ? maxPrice : null,
+      minPrice: minPrice !== currentSearchStats.priceMin ? minPrice : null,
+      maxPrice: maxPrice !== currentSearchStats.priceMax ? maxPrice : null,
     };
-    this.closePopup(() => {
-      this.props.onChange(filter);
-      Logger.hotelsFilterTagSet('Price');
-    });
-  };
-
-  getTitle = (
-    start: number,
-    end: number,
-    min: number,
-    max: number,
-    currency: string,
-    daysOfStay: number,
-  ) => {
-    if (start === min && end === max) {
-      return <Translation id="hotels_search.filter.price_filter.price" />;
-    }
-    if (start === min) {
-      return (
-        <TranslationFragment>
-          <Translation passThrough="< " />
-          <Price amount={end * daysOfStay} currency={currency} />
-        </TranslationFragment>
-      );
-    }
-    if (end === max) {
-      return (
-        <TranslationFragment>
-          <Translation passThrough="> " />
-          <Price amount={start * daysOfStay} currency={currency} />
-        </TranslationFragment>
-      );
-    }
-    return (
-      <TranslationFragment>
-        <Price amount={start * daysOfStay} currency={currency} />
-        <Translation passThrough=" - " />
-        <Price amount={end * daysOfStay} currency={currency} />
-      </TranslationFragment>
-    );
-  };
-
-  render() {
-    const {
-      currentSearchStats: { priceMin, priceMax },
-      isActive,
-      currency,
-      daysOfStay,
-    } = this.props;
-    const start = this.props.start || priceMin;
-    const end = this.props.end || priceMax;
-    if (daysOfStay === null) {
-      return null;
-    }
-    return (
-      <React.Fragment>
-        <FilterButton
-          title={this.getTitle(
-            start,
-            end,
-            priceMin,
-            priceMax,
-            currency,
-            daysOfStay,
-          )}
-          icon={<Icon name="money" />}
-          isActive={isActive}
-          onPress={this.filterButtonClicked}
-        />
-        <PricePopup
-          isVisible={this.state.isPopupOpen}
-          onClose={this.closePopup}
-          onSave={this.handleSave}
-          min={priceMin}
-          max={priceMax}
-          start={start}
-          end={end}
-          currency={currency}
-          daysOfStay={daysOfStay}
-        />
-      </React.Fragment>
-    );
+    props.onChange(filter);
```

Comment:
If I read well, there is no `closePopup` anymore in the `handleSave` function. Is that intentional?

Reply:
Yes, we always got a warning saying you can not setState after component is unmounted for this filter. I finally figured out why. When we set the new value to context, query component re-renders and searches with new input. Causing this component to unmount. So, there is no need to call closePopup, the component unmounts.